src/analysis/scoping_analysis.cpp

```diff
@@ -130,8 +132,18 @@ struct ScopingAnalysis::ScopeNameUsage {
     StrSet got_from_closure;
     StrSet passthrough_accesses; // what names a child scope accesses a name from a parent scope
 
+    // `import *` and `exec` both force the scope to use the NAME lookup
+    AST_ImportFrom* nameForcingNodeImportStar;
+    AST_Exec* nameForcingNodeBareExec;
+    bool hasNameForcingSyntax() { return nameForcingNodeImportStar != NULL || nameForcingNodeBareExec != NULL; }
+
+    // If it has a free variable / if any child has a free variable
```

Comment:
Can you add a comment with your explanation of why these are needed and how they are not derivable from the other attributes?